### PR TITLE
refector e2e for removing dependency.

### PR DESCRIPTION
**Pull request by newera-begins**
### Which issue this PR addresses:

Please include a link to the ADO work item as well as any GitHub issues.
https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10943719

Fixes

### What this PR does / why we need it:

Refactor E2E "Scale Nodes" test to Use go-client Instead of ARO Client

### Test plan for issue:

How did you test that this PR works?

create local RP cluster.
export CLUSTER=#CLUSTERNAME; go test ./test/e2e -tags e2e,codec.safe -c -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=7cc9e51" -o e2e.test
./e2e.test

### CODE REVIEW

File: test/e2e/scalenodes.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ var _ = Describe("Scale nodes", func() {
 	// nodes to settle after scale down
 	Specify("node count should match the cluster resource and nodes should be ready", func() {
 		ctx := context.Background()
-
+/*
```
Comment (m1kola) [inline test/e2e/scalenodes.go:31]: No reason to keep the old code

File: test/e2e/scalenodes.go [hunk 2]
```diff
@@ -43,7 +43,21 @@ var _ = Describe("Scale nodes", func() {
...
+		// New One
+		machinesets, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			log.Warn(err)
+		}
+		expectedNodeCount := 3 // for masters
+		for _, machineset := range machinesets.Items {
+			expectedNodeCount += int(*machineset.Spec.Replicas)
+		}
+		// New One
```
Comment (m1kola) [inline test/e2e/scalenodes.go:58]:
```suggestion
```

File: test/e2e/scalenodes.go [hunk 3]
```diff
@@ -43,7 +43,21 @@ var _ = Describe("Scale nodes", func() {
 			}
 			expectedNodeCount += int(*wp.Count)
 		}
-
+*/
+		
+		
+		// New One
```
Comment (m1kola) [inline test/e2e/scalenodes.go:49]:
```suggestion
```

**Review by m1kola — CHANGES_REQUESTED** [review]
Few suggestions on how to clean up the code, but the change looks good. Thanks.

> ### Test plan for issue:
>
> How did you test that this PR works?
>
> create local RP cluster.
> export CLUSTER=#CLUSTERNAME; go test ./test/e2e -tags e2e,codec.safe -c -ldflags "-X > github.com/Azure/ARO-RP/pkg/util/version.GitCommit=7cc9e51" -o e2e.test
> ./e2e.test

This is good that you tested locally. I normally do not run e2es locally: I create a PR (sometimes in draft state if I want to test early) and let our CI do the job of testing.

File: test/e2e/scalenodes.go [hunk 4]
```diff
@@ -43,7 +43,21 @@ var _ = Describe("Scale nodes", func() {
...
+		machinesets, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			log.Warn(err)
```
Comment (bryanro92) [inline test/e2e/scalenodes.go:52]: should we maybe error out here? otherwise the below code may also fail
```
		Expect(err).NotTo(HaveOccurred())
```

**Review by bryanro92 — CHANGES_REQUESTED** [review]
Thanks for fixing this issue. This looks good so far. I like your logic of getting replicas for each machine set for total number of workers. just two minor nits: 

- delete the old, commented out code
- consider failing if we err on retrieving the machinesets.

File: test/e2e/scalenodes.go (hunk 4 above)
Comment (m1kola) [reply test/e2e/scalenodes.go:52]: Good catch!

File: test/e2e/scalenodes.go [hunk 5]
```diff
@@ -28,22 +28,13 @@ var _ = Describe("Scale nodes", func() {
 	// nodes to settle after scale down
 	Specify("node count should match the cluster resource and nodes should be ready", func() {
 		ctx := context.Background()
-
-		oc, err := clients.OpenshiftClustersv20200430.Get(ctx, vnetResourceGroup, clusterName)
+		machinesets, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
 		Expect(err).NotTo(HaveOccurred())
-
+
```
Comment (m1kola) [inline test/e2e/scalenodes.go:33]: Need to remove tabs. Linter will probably complain on this one.

I recommend checking editor settings: most editors (vscode is one) allow removing whitespaces at the end of the line automatically. Very useful.

```suggestion

```

**Review by m1kola — DISMISSED** [review]
Looks good to me.

Happy to merge once tests are green.

/azp run ci
/azp run e2e

File: test/e2e/scalenodes.go (hunk 4 above)
Comment (newera-begins) [reply test/e2e/scalenodes.go]: i changed that accordingly,

**Review by bryanro92 — APPROVED** [review]
lgtm

**Review by bennerv — APPROVED** [review]
LGTM

**Pull request merged**